src/box/lua/net_cluster.lua

```diff
@@ -0,0 +1,466 @@
+    end
+    self._watchers[watcher] = watcher
+    return watcher
+end
+
+-- Trigger internal load of config (in case of `config_mode` == 'auto')
+-- set new cluster config (in case of `config_mode` == 'manual')
+-- In the first case `config` must be nil, in the second - it must be
+-- a map replicaset_name -> replicaset_cfg (table).
+-- replicaset_cfg may have members:
+--  connect_cfg: if provided, it will be passed to net.replicaset.connect.
+--   (if not, `replicaset_name` will be passed along with connect_opts).
+--  roles: optional list of roles (map or array) that the replicaset have.
+--   if present, affects filtering by role in some methods.
+function cluster_methods:reload_config(config)
+    cluster_check(self, 'reload_config')
+    if config then
+        if self._cfg.config_mode ~= 'manual' then
+            box.error(box.error.ILLEGAL_PARAMS,
+                    'manual reload expects manual config mode')
+        end
+        reload_config_impl(self, config)
+    else
+        if self._cfg.config_mode ~= 'auto' then
+            box.error(box.error.ILLEGAL_PARAMS,
+                    'force reload expects auto config mode')
+        end
+        reload_config_impl(self, collect_config())
+    end
+end
+
+-- Close all watchers and connections.
+function cluster_methods:close()
+    cluster_check(self, 'close')
+    while true do
+        local watcher = next(self._watchers)
+        if not watcher then
+            break
+        end
+        watcher:unregister()
+    end
+    for _, replicaset in pairs(self._replicasets) do
+        close_replicaset_connection(replicaset)
+    end
+    table.clear(self._replicasets)
+    if self._reload_watcher then
+        self._reload_watcher:unregister()
+        self._reload_watcher = nil
+    end
+    self._is_closed = true
+end
+
+-- get for default cluster (which is created if needed).
+local function get(replicaset_name)
+    if not default_cluster then
+        default_cluster = cluster_new(cluster_cfg_defaults)
+    end
+    return default_cluster:get(replicaset_name)
+end
+
+-- get_connections for default cluster (which is created if needed).
+local function get_connections(opts)
+    if not default_cluster then
+        default_cluster = cluster_new(cluster_cfg_defaults)
+    end
+    return default_cluster:get_connections(opts)
+end
+
+-- watch_leaders for default cluster (which is created if needed).
+local function watch_leaders(key, func, opts)
+    if not default_cluster then
+        default_cluster = cluster_new(cluster_cfg_defaults)
+    end
+    return default_cluster:watch_leaders(key, func, opts)
+end
+
+-- reload_config for default cluster (which is created if needed).
+local function reload_config(config)
+    if not default_cluster then
+        default_cluster = cluster_new(cluster_cfg_defaults)
+    end
+    return default_cluster:reload_config(config)
+end
+
+return {
+    new = cluster_new,
+
+    get = get,
+    get_connections = get_connections,
```

Comment:
I'm in a doubt about these methods/functions.

First, I'm not sure about names. `get` looks like something to be frequently used, because of the short/common word name. `get` and `get_connections` feels non-uniform: just 'get' and 'get something' for a same-level methods.

Huh, choosing names is the hardest task in programming :)

Second, I'm not sure about exposing `net.replicaset` objects from the a-kind-of-public API, while the `net.replicaset` module is internal.

Also, if we have a more general connections-by-filters method, should we also provide a particular connections-by-replicaset-name method? (I assume that a filter by a replicaset name may be included into the list of the filters.)

If it worth such a shortcut, how about an approach that we use in the [testing helper](https://github.com/tarantool/tarantool/blob/2467e421cbfd42d0f58c0d27e4e397e730412e6c/test/config-luatest/cluster.lua#L18) for working with a cluster: get a replicaset/instance representation (connection) using the indexing operator? Sure, it technically forbids using of some instance/replicaset names, but it doesn't look as a problem in practice.

These are a bit too picky comments considering the experimental status of the module. I left them to share thoughts and I don't want them to be blockers.

We discussed it with @alyapunov and he said that he can replace these two methods with one `:call_leader(replicaset_name, <...>)` (or something of this kind). It seems it would solve my concerns.